terraform/environments/data-platform/lambda_for_glue_metadata.tf

```
@@ -0,0 +1,120 @@
+data "archive_file" "get_glue_metadata_zip" {
+  type        = "zip"
+  source_dir  = "${path.module}/src/get_glue_metadata"
+  output_path = "${path.module}/src/get_glue_metadata_${local.environment}/get_glue_metadata_lambda.zip"
+}
+
+resource "aws_iam_role" "get_glue_metadata_lambda_role" {
+  name               = "get_glue_metadata_role_${local.environment}"
+  assume_role_policy = data.aws_iam_policy_document.lambda_trust_policy_doc.json
+  tags               = local.tags
+}
+
+data "aws_iam_policy_document" "iam_policy_document_for_get_glue_metadata_lambda" {
+  statement {
+    sid       = "GlueReadOnly"
+    effect    = "Allow"
+    actions   = ["glue:Get*"]
```

```suggestion
    actions   = ["glue:GetTable"]
```
I assume this is the only permission needed to call the get_table endpoint in boto3; `glue:GetDatabase` might also be needed